src/unstract/sdk/file_storage/fs_permanent.py

```diff
@@ -45,8 +46,9 @@ def _copy_on_write(self, path):
         """
         if not self.exists(path):
             local_file_storage = FileStorage(provider=FileStorageProvider.Local)
-            if local_file_storage.exists(path):
-                self.fs.put(path, path)
+            local_file_path = self.legacy_storage_path + "/" + path
```

Comment:
NIT: I suggest using Pathlib to perform any path based manipulation rather than directly manipulating as strings

Reply:
I agree. However, I do not see many use cases where we do this for now. Also this is a piece which can be removed later as this is only to support legacy migration for now.... Hence leaving it for now. Hope that is ok.